### PR TITLE
Sign in

### PR DESCRIPTION
**Pull request by kevin-bermudez-wolox**
## Summary

- Add sign in service

## Known Issues

None

## Trello Card

https://trello.com/c/Gb3Eqfs7/2-sign-in

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -21,3 +26,25 @@ exports.signUp = async (req, res, next) => {
...
+  try {
+    const userData = await mapperSignIn(req.body);
+    const existUser = await getUserByEmail(userData.email);
+
+    if (!existUser) {
+      return next(unauthorizedError('User not found'));
+    }
+
+    if (!verify(userData.password, existUser.password)) {
+      return next(unauthorizedError('User not found'));
```
Comment (dbenavidesv) [inline app/controllers/users.js]: Use a different error message

File: app/schemas/sign_in.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    .withMessage('must be a valid email'),
+  body('email')
+    .trim()
+    .notEmpty()
+    .exists()
+    .withMessage('is required'),
+  body('password')
+    .isAlphanumeric()
+    .isLength({ min: 8 })
+    .withMessage('must be alphanumeric and have a minimum of 8 characters'),
```
Comment (dbenavidesv) [inline app/schemas/sign_in.js]: As this is a sign in service, maybe don't send this specific message, `is not valid` would be enough

File: app/serializers/users.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+exports.signIn = token => ({ token });
```
Comment (dbenavidesv) [inline app/serializers/users.js]: Just to comment, serializers are not always necessary

File: app/services/users.js [hunk 4]
```diff
@@ -27,3 +28,13 @@ exports.createUser = async ({ name, lastName, email, password }) => {
     throw databaseError('Error signUp user');
   }
 };
+
+exports.createToken = ({ id }) => {
```
Comment (dbenavidesv) [inline app/services/users.js]: This should be a helper function, not a service

**Review by dbenavidesv — APPROVED** [review]

**Review by jxczz2 — APPROVED** [review]

**Pull request merged**